### PR TITLE
Fix invalid output for record with accessibility modifier & members

### LINKED ISSUE #2511: Stroustrup record type definition with access modifiers and members lead to invalid F# code
Issue created from [fantomas-online](https://fsprojects.github.io/fantomas-tools/#/fantomas/preview?data=N4KABGBEDGD2AmBTSAuKAXAngB0WAcrAHYCiAttlgDICWAzugDwDkAKgHxgC8AOkRBGwAnGgDcAhukR8BssMDC0GaNmAA29dGAC%2BfGQLKIyAI0RCw6ABb0AdAAlE4%2BNwvW6NpenuP4%2BiIZMzV1tWcRo1FytbTxtQ8L8wANNzKPcqRCIAcytItw9NDwzsyz5IABpwKDpEdHQaLLpUMABtSohQOSgAEixcJsh69HK2gUguiTUAV2Q0Vs6IAAYK%2BagASSIkInQAZRoAL2Rl%2BYAWEYgAXRHtI-az7t6ZqEHhzrGJ6aa5%2BYBGG7lIACy4gAHrQiIh0lkrC8fgAmBZ3S6ya4jDr-Ho4R6QDLwADyADMwYhtlg1Ic7m9xFNHl9OrC-rJICQNgSiTDXmp8ZBEVc-mjGRi%2BmhIMZYLA1OzRuMqR9ZncIABmBmjdbVIToABi9Sp%2BEQAHcNODJbJ0EJpjzkXyKYKsaLxcbuu8afKwMdlRBINtsOJoIgAEKIfGwISIAAK4iE4kMUiEDogpvNnSRAhRsn5Uoe-TtEuVlOpnxdAFZ3VAvT7-YHgxDYHqzNBxNV1qJYPW6sQ4xYzYgLSmra8bVmxTnrU6CysAGwlz3e30BoMhgCq2FwQnrjaIzdbNHbJfxVOqPYgqYE6Y9A%2BF2YdedlLRdAHYp2XZ5WQwBhNQNuiv4gMM3QdDBh2e5qAeSa8qi1qZheQ5XtK%2BZyisAAcj4zhW86IACRjJEB%2B7dmBloQf2UFQJeuZwTetJyAAnCh5ZzlW35qDuLrAaBcjJkefbosRIowWRo4IT8SwutO5YAIL4jG35kGQ4gdgmeHseBaaQZig72vxMrOis3y-CJT5oVW2xGDQcBMUQOEgYpsgcToXECjxpEjlpY5wrRvoSTGxlkKZ4rMSsCmHnZhHcWp0Eac58G3jpSr6ahYlCLAkwbAAIogGg%2BTG8ldkFx63ERYVPFssECdFPxuiJQLAqs%2BKsJYGTbJYwboAA6jQ8DQiWCL4b2IUOYVAzFZpUWUbI3zFpVII1XVGQkFZjXNW1HWWB247dUpBEqQVQpFUMw0US63yTpN1VEPiNDAriK6SMGJDAsIiB0HQ24WSWiHrTZyknqpO2DXtkUHTpD4nQASogcBCPAS2dS6xwfQItl5fIP1Ys8%2B3aT8yGg%2BDwbwPgkyBEIBKrFIZCNCW3y5fZGYDWQkxqHUhqIBqwaybUZisGp6OuZ03w0SJYMQ-AAL04z9TM6zkhZVO0CWBGPoxgA%2Bnq7XQlTfU079aMAxjdLCSsgIgglkaYLiQieNDK1vfDFxffloVa0NOs83IsJ6QbVXG%2BIpvm5o%2BOE8TpPk4d6tbQ7WJ0wz4TiyzQhszGnN9Nzgl0vSIlez7ngi1HTOx-HZgdjAcuRv%2BZjK6rK2h9922o07-alaNAiwrFHsggAalpfr1PA9SZJbHbvVX9v9Y7-31y5KeuxVrfAhqyX-i9XcbL3-clnDQ-IzX-Ta%2BPI0urCE0zylsDoAA4jUd0PU9L2ry6g89ZxGtno5fHO5PsiwsdHui9H4J%2Bkx0AADWfoS6AJqHQXERBthRkQIxAmr0XSBQfsFMOI9bSv13oDeYsJgYG11AacWAZ0C1gyInRAaVzpEBoG2IgYkNiYUJsHAKOVkFI1PPcAaTlMG61dljA2YkNCZCIHPIgC9iC7CEZISYIZWCwHWJsdAkgXqWTYp9Ta1dw7qWHNwl2H9%2Bb8IZmYIgSjRDViyAw5IFD6jUJekw%2BYrFrIIztpvTR4VtHogbi6BU%2Bt5iAh-kSKgUZjDwHEO%2BWAz0sj4KZioxxtt1HD01ugiKOj34CAVO7Xxl8zA0EMFsKkABpRAiBsDyIyOgdYIDxCiKtixXCG92FjBfskjxE8yqdAVGnA2-8qmAKJHQBKSUNi6gYIgYWP8maXxDNfH82VEwbV6qgxJWiSqtMboqFuvi-QRnoiGFK9BoAiB8sYqQ8AFxUOIGlaAH5Iw0JifUlGyzk5tLkAqaemT7rZNyYotQJJEqTF-JMbAJJMBklmbEsAiNqbPwGjvFpe8VgKkPr4wpxSqr%2B2SASbpRBenizsbzBYNsIXOIaeeEiGC4VYPaV-XxvzTLoABAgckKwHG5UqOcI4Aw6Aamek0Fl2ggA)

#### Code

```fsharp
type NonEmptyList<'T> =
    private
        { List: 'T list }

    member this.Head = this.List.Head
    member this.Tail = this.List.Tail
    member this.Length = this.List.Length

```
            

#### Result

```fsharp
type NonEmptyList<'T> =
    private {
        List: 'T list
    }

member this.Head = this.List.Head
member this.Tail = this.List.Tail
member this.Length = this.List.Length

```
            
#### Problem description

Please describe here the Fantomas problem you encountered.
                    Check out our [Contribution Guidelines](https://github.com/fsprojects/fantomas/blob/master/CONTRIBUTING.md#bug-reports).

#### Extra information

- [x] The formatted result breaks my code.
- [ ] The formatted result gives compiler warnings.
- [ ] I or my company would be willing to help fix this.

#### Options

Fantomas master branch at 2022-09-17T05:39:07Z - c1a99d498361c19f4e66b8877dd237275168d63d

```fsharp
    { config with
                MultilineBlockBracketsOnSameColumn = true
                ExperimentalStroustrupStyle = true }
```

<sub>Did you know that you can ignore files when formatting from fantomas-tool or the FAKE targets by using a [.fantomasignore file](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#ignore-files-fantomasignore)?</sub>

### PR DESCRIPTION
**Pull request by josh-degraw**
Closes #2511

### CODE REVIEW

**Comment by josh-degraw** [comment]
I fixed the invalid output, but it also brought up another Stroustrup style-guide question: https://github.com/fsharp/fslang-design/issues/706#issuecomment-1250134427

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Fixed
 * Items.Item \<index> <- value gives wrong code. [#2498](https://github.com/fsprojects/fantomas/issues/2498)
+Fix invalid output for record with accessibility modifier & members [2516](https://github.com/fsprojects/fantomas/pull/2516)
```
Comment (nojaf) [inline CHANGELOG.md:7]: Please don't use the PR link but the issue title and link instead.
Also, start the entry with `*`.

**Review by nojaf — APPROVED** [review]
Thanks, you are on a roll!

**Comment by josh-degraw** [comment]
> Thanks, you are on a roll!
> 

It's easy when you tell me exactly how to do it lol 😅

**Pull request merged**